macros/metrics/base/expression.sql

```diff
@@ -40,36 +41,70 @@
     {% do metrics_to_compute.extend(var('re_data:metrics_base')['table']) %}
     {% do metrics_to_compute.extend(metrics.get('table', [])) %}
 
-    {% for metric in metrics_to_compute %}
-        {% set expression = re_data.metrics_base_expression_table(time_filter, metric) %}
-        {% do table_expr.append({ 'expr': expression, 'col_name': '', 'metric': metric}) %}
+    {% for metric_value in metrics_to_compute %}
+        {% set metric_obj = re_data.extract_metric_config(metric_value) %}
+        {% set expression = re_data.metrics_base_expression_table(time_filter, metric_obj['metric'], metric_obj['config']) %}
+        {% do table_expr.append({ 'expr': expression, 'col_name': '', 'metric': metric_obj['metric']}) %}
     {% endfor %}
 
     {{ return (table_expr) }}
 
 {% endmacro %}
 
-{% macro metrics_base_expression_table(time_filter, metric) %}
+{% macro metrics_base_expression_table(time_filter, metric, config) %}
     {% set macro_name = 're_data_metric' + '_' + metric %}
+    {% set metric_macro = re_data.get_metric_macro(macro_name, metric) %}
 
-    {% if context['re_data'].get(macro_name) %}
-        {{ context['re_data'][macro_name](time_filter) }}
+    {% if config is not none %}
+        {{ metric_macro(time_filter, config) }}
     {%- else %}
-        {{ context[project_name][macro_name](time_filter) }}
+        {{ metric_macro(time_filter) }}
     {% endif %}
 
 {% endmacro %}
 
 
-{%- macro metrics_base_expression_column(column_name, metric) %}
+{%- macro metrics_base_expression_column(column_name, metric, config) %}
     {% set macro_name = 're_data_metric' + '_' + metric %}
```

Comment:
`macro_name` this is actually also initialized in `re_data.get_metric_macro`. So here it's no longer needed

Reply:
Yeah that's right